src/core/core.scale.js

```diff
@@ -302,20 +316,21 @@ var Scale = Element.extend({
 
 		// IMPORTANT: from this point, we consider that `this.ticks` will NEVER change!
 
-		// BACKWARD COMPAT: synchronize `_ticks` with labels (so potentially `this.ticks`)
-		for (i = 0, ilen = labels.length; i < ilen; ++i) {
-			label = labels[i];
-			tick = ticks[i];
-			if (!tick) {
-				ticks.push(tick = {
-					label: label,
+		if (!ticks) {
+			ticks = [];
+			for (i = 0, ilen = labels.length; i < ilen; ++i) {
+				ticks.push({
+					value: me.ticks[i],
```

Comment:
I think I'd set `label` here too, and put next loop in `else`, to avoid having 2 loops.

Reply:
I updated a bit differently. The loops have different `ilen` now, which makes them harder to combine, but I combined `if` checks instead.

I had separated because it's helpful to my upcoming change of generating labels after autoSkip. In that case I'll need to create the ticks, then skip, then set labels, so it was useful to separate tick creation and setting labels. This change might make more sense as part of that PR if you'd like me to defer the change